src/reason-parser/reason_pprint_ast.ml

```diff
@@ -2368,13 +2368,29 @@ let constant_string f s = pp f "%S" s
 
 let tyvar f str = pp f "'%s" str
 
+let pp_string_literal f s =
+  let buf = Buffer.create 4 in
+  Uutf.String.fold_utf_8 (
+    fun () _i uc ->
+      match uc with
+      | `Malformed s -> failwith (Format.asprintf "Invalid UTF-8 string literal: %s" s)
```

Comment:
The string doesn't have to be valid utf. it seems wrong to fail here.
one can easily write a string litteral such as "\xc3\x28"

Reply:
Agreed - `failwith` is not meant to be permanent.  The current PR really is meant to be a minimal start to how something like this could work.